plugins/modules/source_control/gitlab/gitlab_protected_branch.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: gitlab_protected_branch
+short_description: Creates/deletes proteched branches in Gitlab for an project.
+description:
+  - Create/Delete a protected branch in an project.
+author:
+  - "Werner Dijkerman (@dj-wasabi)"
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+extends_documentation_fragment:
+- community.general.auth_basic
+
+options:
+  state:
+    description:
+      - Create or delete proteced branch.
```

Comment:
```suggestion
      - Create or delete protected branches.
```
Is this actually about protecting and unprotecting branches, or creating and deleting branches (which happen to be protected)? (If this is about actually creating branches, you need to also provide a SHA1 or branch name where to create the branch from.)

Reply:
The first.

Reply:
In that case, you should definitely adjust the documentation, because right now it says the other :)

Reply:
I don't understand, what should I adjust then? Because in Gitlab it is called a protected branch and not sure what I should do then? It is about creating protected branches or not.

Reply:
"Protect or unprotect branches." is very different from "Create and delete protected branches.". The module does **not** create protected branches, it marks existing branches as protected (or in other words: it creates a branch protection).